Add model creator, test data and tests

sandertan opened the pull request:
Hi @w-is-h This PR adds a model creator based on the MedCAT tutorials. This makes it easy to create CDB and Vocab files from a concept csv, text file for unsupervised training, and a set of parameters. Discussed in #116 

Let me know what you think. If necessary we can implement changes to fit it more into the existing codebase.

## Added in this PR
- Model creator
- Test data
- Test for creating CDB & Vocab
- Test for basic entity linking from CDB
- Test for entity linking using some specific configurations

## To do in this PR
- [x] Add documentation on how to run the pipeline (`python create_models.py <path_to_config.yml>`and use the config. We can add a section in the README above the `#Model` section.
- [x] Decide on config: use a new config file or can we incorporate in existing?
- [x] Check naming of config properties and variables. Are they consistent with existing code?
- [x] Decide on tests:
	- Is format okay? It's in different format compared to existing tests
	- ~Keep as single file or split into multiple? The 2nd Class is dependent on the 1st because it creates CDB and Vocab. We can also include a CDB and vocab to make them independent.~
- [x] Verify automated tests are functioning.
- [x] Use HF comment style
- [x] Squash & rebase before merging

## To do at later time
We're currently not using these functionalities so might be better to let someone else implement this):
- Add functionality for optional supervised training
- Add functionality for multiple concept csvs
- Add functionality for multiple unsupervised training data files
- Add functionality to output model pack
- Add functionality to the logger to use properties defined in general config

baixiac reviewed (commented):
LGTM. IMHO the `model_creator` could go to the utils/helpers module.

sandertan commented:
@baixiac thanks for the review. I'll try to implement the changes you requested this week.

sandertan commented:
Hi @baixiac I've implemented the changes you requested. In summary:
- I've split tests to assert a single scenario per test. That increased the number of green checkmarks :) I did group the tests in classes, still in the same file, which clarifies what the grouped tests are about.
- I've moved the model creator to the utils folder.
- I've added documentation on how to use the model creator, and including links to examples and existing documentation.

Before merging, I should rebase it on master since I'm not sure it still works with the recent changes. A future enhancement could be to output the model pack instead of separate CDB and Vocab files. Are there any other things I should add to this PR? And shall I squash the commits?

sandertan commented:
Another future enhancement could be to improve the logger. For example, use the output format defined in the medcat config file.

baixiac reviewed (approved):
Thanks for the contribution. Agreed and the logger should have loaded the general config. Squash and merge will help you to summarise the change indeed.

Do you wanna have a final look before the merge? @w-is-h

w-is-h commented:
@baixiac i trust all is good

sandertan commented:
@baixiac Squashed & rebased.
